src/main/scala/com/raphtory/core/components/partition/Reader.scala

```diff
@@ -111,8 +113,10 @@ class Reader(
                 serialise(WatermarkTime(partitionID, finalTime, noBlockingOperations))
         )
         lastWatermark = (finalTime, noBlockingOperations)
+        PartitionTelemetry.lastWaterMarkProcessed.set(finalTime)
```

Comment:
Does this manage global watermarks i.e. take into account the possibility of multiple readers?

Reply:
This uses partition ID in it's builder, hence this would show last watermark processed per partition id